### DISCUSSION TITLE
Add absolute path for executables like singularity & others

### DISCUSSION THREAD

**Discussion by mattheatley** [nextflow-io/nextflow#3413] [Q&A]
at our institution we have different executables for singularity ('singularity' and 'singularity-mod' -  the latter is a bash script that runs 'umask 0022' prior to singularity which ensures that docker containers are reproduced correctly). I couldn't see any option to customise which 'singularity' is used for pull / exec and this would be useful, probably in the .config file.

**Comment by martinfthomsen** [comment]
Hi Matt,
There is the `singularity.cacheDir = '/path/to/container/directory'` option that can be set inside a config file
or you can set the `$NXF_SINGULARITY_CACHEDIR` environment variable you can set:
`export NXF_SINGULARITY_CACHEDIR='/path/to/container/directory'`
Have you tried these?

**Comment by mattheatley** [comment]
hi thanks for the quick response!

It's not the cache directory that is an issue but the path to the executable 'singularity' that actually runs the software i.e. when running 'singularity pull ...' or 'singularity exec ...' etc.

as i understand it, at the moment nextflow just searches for the 'singularity' executable in $PATH but it would be great if there could be an option to specify the path for the executable (and others) instead.

i'm hopful that i'll be able to find a solution to an issue related to this (largely due to our institutional setup) but thought it could be a useful addition in future versions
